src/main/java/hillelee/homework/doctor_api/DoctorService.java

```diff
@@ -0,0 +1,32 @@
+package hillelee.homework.doctor_api;
+
+import lombok.AllArgsConstructor;
+import org.springframework.stereotype.Service;
+
+
+/**
+ * Created by 4oc3p on 30.11.2017. hillelee
+ */
+@Service
+@AllArgsConstructor
+public class DoctorService {
+    DoctorRepository doctorRepository;
+
+    public void addDoctor(Doctor doctor) {
+        Integer doctorId = doctorRepository.getDoctors().keySet().stream()
+                .mapToInt(Integer::valueOf)
+                .max()
+                .orElse(-1);
+        doctor.id = ++doctorId;
```

Comment:
this trick doesn't solve concurrency problem :(
Two requests (threads) can retrieve same ID simultaneously